### DISCUSSION TITLE
How to resize scrolling area when the diagram changes size

### DISCUSSION THREAD

**Discussion by yannicklerestif** [antvis/X6#1913] [Q&A] [answered]
Hello all!

I'm wondering how to resize the scrolling area when the diagram is resized.

I'm attaching a screen capture video so you can see what I mean:
![x6-resize](https://user-images.githubusercontent.com/5543284/159266336-14697b18-e2a7-4b10-8427-122ad42a1509.gif)

Here is a code sandbox with the code I'm using: https://codesandbox.io/s/x6-test-unw5yf

In the capture above, I would like the scrolling area to always cover the whole blue area.

Right now I'm [using Window resize event to manually resize the graph](https://codesandbox.io/s/x6-test-resize-workaround-spiimg?file=/src/app.tsx:528-877), which does work, but I'm wondering if there is a better way.

Thanks in advance!

Yannick

**Comment by NewByVector** [comment]
Using window resize event to resize graph seems to be a good way to do it.

**Comment by yannicklerestif** [comment, marked as answer]
Ok thanks!
